subworkflows/local/transcript_identification/main.nf

```diff
@@ -0,0 +1,117 @@
+include { BEDTOOLS_SORT                                   } from '../../../modules/nf-core/bedtools/sort/main'
+include { BEDTOOLS_INTERSECT as BEDTOOLS_INTERSECT_FILTER } from '../../../modules/nf-core/bedtools/intersect/main'
+include { BEDTOOLS_INTERSECT                              } from '../../../modules/nf-core/bedtools/intersect/main'
+
+workflow TRANSCRIPT_INDENTIFICATION {
+    take:
+    group_bam_bai
+    gxf
+    fasta
+    chrom_sizes
+
+    main:
+
+    ch_versions = Channel.empty()
+    ch_identification_bed = Channel.empty()
+
+    grohmm_td_plot = Channel.empty()
+    if (!params.skip_grohmm && params.assay_type == "GROseq") {
+        GROHMM(group_bam_bai, gxf)
+        ch_identification_bed = ch_identification_bed.mix(GROHMM.out.bed)
+        grohmm_td_plot = GROHMM.out.td_plot
+        ch_versions = ch_versions.mix(GROHMM.out.versions.first())
+    }
+
+
+    homer_peaks = Channel.empty()
+    homer_tagdir = Channel.empty()
+    if (params.assay_type == "GROseq") {
+        group_bam = group_bam_bai.map { meta, bam, bai -> [meta, bam] }
+        HOMER_GROSEQ(group_bam, fasta)
+        ch_identification_bed = ch_identification_bed.mix(HOMER_GROSEQ.out.bed)
+        homer_peaks = HOMER_GROSEQ.out.peaks
+        homer_tagdir = HOMER_GROSEQ.out.tagdir
+        ch_versions = ch_versions.mix(HOMER_GROSEQ.out.versions.first())
+    }
+
+    // Scatter the chromosomes
+    // TODO Might turn this into a param
+    skip_chr = ["chrY", "_random", "chrUn_", "chrEBV", "chrM"]
+    ch_chr = fasta
+        .splitFasta(record: [id: true])
+        .map { record -> record.id }
+        .filter { !(it in skip_chr) }
+
+    // NOTE https://github.com/hyulab/PINTS/issues/15
+    // We want to run PINTS once per biological/technical sample.
+    // Let Nextflow handle the parallelization Per the docs
+    // Per the docs:
+    // {SID} will be replaced with the number of samples that peaks are called from,
+    // if you only provide PINTS with one sample, then {SID} will be replaced with 1,
+    // if you try to use PINTS with three replicates (--bam-file A.bam B.bam C.bam),
+    // then {SID} for peaks identified from A.bam will be replaced with 1.
+    SAMTOOLS_MERGE(
+        group_bam_bai.map { meta, bams, _bais -> [meta, bams] },
+        [[], []],
+        [[], []]
+    )
+    ch_versions = ch_versions.mix(SAMTOOLS_MERGE.out.versions.first())
+
+    PINTS_CALLER(
+        SAMTOOLS_MERGE.out.bam.combine(ch_chr),
+        params.assay_type
+    )
+    ch_versions = ch_versions.mix(PINTS_CALLER.out.versions.first())
+
+    // Gather the chromosomes
+    // TODO Tests don't seem to hit this because there's no bidirectional_TREs
+    // Need to collect all of the beds for each chromosome/sample and concatenate them
+    // Nextflow makes this super easy
+    def ch_bidirectional_TREs = PINTS_CALLER.out.unidirectional_TREs.groupTuple(by: [0]).map { meta, beds ->
```

Comment:
why use `def` here ?

Reply:
The new extension told me to do it 🤷🏻‍♂️ If you turn on the future warnings, I think?